src/structures/WelcomeScreen.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+const { default: Collection } = require('@discordjs/collection');
+const Base = require('./Base');
+const WelcomeChannel = require('./WelcomeChannel');
+
+/**
+ * Represents a welcome screen.
+ */
+class WelcomeScreen extends Base {
+  constructor(guild, data) {
+    super(guild.client);
+    /**
+     * The guild for this welcome screen
+     * @type {Guild}
+     */
+    this.guild = guild;
+
+    this._patch(data);
+  }
+
+  /**
+   * Builds the welcome screen with the provided data.
+   * @param {*} data The raw data of the welcome screen
+   * @private
+   */
+  _patch(data) {
+    if (!data) return;
+
+    /**
+     * The description of this welcome screen
+     * @type {?string}
+     */
+    this.description = data.description ?? null;
+
+    if (!this.welcomeChannels) {
+      /**
+       * Collection of welcome channels belonging to this welcome screen
+       * @type {Collection<Snowflake, WelcomeChannel>}
+       */
+      this.welcomeChannels = new Collection();
+    }
```

Comment:
Maybe this could be moved into the constructor instead of checking if it already exists?

Reply:
took that approach from another structure that does it, might happen if we decide this needs to be patchable via update gateway events further down the line